kore/src/Kore/Step/Simplification/Predicate.hs

```diff
@@ -27,102 +25,57 @@ import Kore.Internal.Pattern
     , Predicate
     )
 import qualified Kore.Internal.Pattern as Pattern
+import qualified Kore.Internal.Predicate as Predicate
 import qualified Kore.Predicate.Predicate as Syntax
     ( Predicate
     , unwrapPredicate
     )
 import qualified Kore.Predicate.Predicate as Syntax.Predicate
-    ( substitute
-    )
 import Kore.Step.Simplification.Simplify
 import Kore.Step.Substitution
-    ( mergePredicatesAndSubstitutions
+    ( normalize
     )
 import qualified Kore.TopBottom as TopBottom
-import Kore.Unification.Substitution
-    ( Substitution
-    )
 import qualified Kore.Unification.Substitution as Substitution
 import Kore.Unparser
 
 {- | Create a 'PredicateSimplifier' using 'simplify'.
 -}
-create :: PredicateSimplifier
-create = PredicateSimplifier (simplify 0)
+create :: MonadSimplify simplifier => PredicateSimplifier simplifier
+create = PredicateSimplifier simplify
 
 {- | Simplify a 'Predicate'.
 
 @simplify@ applies the substitution to the predicate and simplifies the
-result. The result is re-simplified once.
+result. The result is re-simplified until it stabilizes.
+
+@simplify@ accepts any @term@, but the 'Conditional' 'term' is not simplified.
```

Comment:
Would renaming the type variable `term` to `a` enforce this comment?

Am I understanding it right that this function couldn't simplify its `term` even if it wanted, because it can't figure out `term ~ TermLike`?

Reply:
That's right; `term` is `forall`-quantified, so it can't be unified with any particular type. I did it this way because it's very common that we want to simplify the predicate of some configuration and it's needlessly awkward to always take the configuration apart, simplify the predicate, and reassemble it—at every call site.